### PR TITLE
Save date company published

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-1022)

### CODE REVIEW

File: company/signals.py [hunk 1]
```diff
@@ -47,3 +47,8 @@ def publish_companies_that_meet_criteria(sender, instance, *args, **kwargs):
             instance.email_address and
             instance.verified_with_code
         )
+
+
+def store_date_published(sender, instance, *args, **kwargs):
+    if instance.is_published and not instance.date_published:
+        instance.date_published = timezone.now()
```
Comment (richtier) [inline company/signals.py]: I considered putting this inside `publish_companies_that_meet_criteria`, but we want to store the date if admin user manually sets the company to published.

File: company/tests/test_migrations.py [hunk 2]
```diff
@@ -8,41 +8,12 @@
 
 
 @freeze_time()
-def test_populate_company_populate_verification_date(migration):
+def example_test(migration):
```
Comment (richtier) [inline company/tests/test_migrations.py]: Once new migrations happen the existing migrations tests no longer work.

I'm keeping the boilerplate as the layout of the migration test is not very intuitive.

File: company/tests/test_migrations.py [hunk 3]
```diff
@@ -1,48 +1,9 @@
...
-
-from freezegun import freeze_time
-
-from django.utils import timezone
-
-from company.tests.factories import CompanyFactory
-
-
-@freeze_time()
-def test_populate_company_populate_verification_date(migration):
```
Comment (konradko) [inline company/tests/test_migrations.py]: please remove, seems like it is not testing anything

Comment (richtier) [inline company/tests/test_migrations.py]: it is now removed, but github does not realise your comment has been reacted to

**Pull request merged**